sale_product_seasonality/models/sale_order.py

```diff
@@ -30,12 +30,19 @@ def _default_seasonal_config_id(self):
 
     def _get_allowed_products(self, date):
         self.ensure_one()
-        products = self.env["product.product"]
+        product_ids = []
         if date:
-            config_lines = self.mapped("seasonal_config_id.line_ids")
-            products = config_lines.filtered(lambda l: l.is_sale_ok(date)).mapped(
-                "product_id"
+            config_lines = self.mapped("seasonal_config_id.line_ids").filtered(
+                lambda l: l.is_sale_ok(date)
             )
+            for config_line in config_lines:
+                if config_line.product_id:
+                    product_ids.append(config_line.product_id.id)
+                else:
+                    product_ids.extend(
+                        config_line.product_template_id.product_variant_ids.ids
+                    )
+        products = self.env["product.product"].browse(product_ids)
```

Comment:
NOTE: we could probably avoid browsing here as the compute method requires IDS

Reply:
Fixed with commit amend/push force: https://github.com/OCA/sale-workflow/compare/9a8f2a52369ca025bf1bc3b69c7213da536183c6..6dd2b5faafcc500a546063af005a454b4fcbb170